STYLEGUIDE_CONVENTION.md

```diff
@@ -0,0 +1,199 @@
+# Code style guide
+
+## Table of Contents
+
+1. [React](#react)
+2. [TypeScript](#typescript)
+3. [CSS](#css)
+
+## React
+
+### JSX
+
+Prefer normal functions (non arrow functions or classes):
+
+```JSX
+// bad
+class Listing extends React.Component {
+  render() {
+    return <div>{this.props.hello}</div>;
+  }
+}
+
+// bad (relying on function name inference is discouraged)
+const Listing = ({ hello }) => (
+  <div>{hello}</div>
+);
+
+// good
+function Listing({ hello }) {
+  return <div>{hello}</div>;
+}
+```
+
+### Extract Login from jsx
+
+```JSX
+//bad
+function FooComponent({ onClick }) {
+ return <button onClick={()=>{
+   ...
+   onClick();
+ }} />
+}
+
+//good
+function FooComponent({ onClick }) {
+ const handleButtonClick = () => {
+  ...
+  onClick();
+ }
+
+ return <button onClick={handleButtonClick} />
+}
+```
+
+### Props Naming
+
+Avoid using DOM component prop names for different purposes.
+
+> Why? People expect props like `style` and `className` to mean one specific thing.
+> Varying this API for a subset of your app makes the code less readable and less maintainable, and may cause bugs.
+
+```JSX
+// bad
+<ExampleComponent style="fancy" />
+
+// bad
+<ExampleComponent className="fancy" />
+
+// good
+<ExampleComponent variant="fancy" />
+```
+
+### Props event naming
+
+```JSX
+// bad
+function ExampleComponent({
+  click,
+  submit,
+  ...
+}) {
+  return ...
+}
+
+// bad
+function ExampleComponent({
+  handleClick,
+  handleSubmit,
+  ...
+}) {
+  return ...
+}
+
+// good
+function ExampleComponent({
+  onClick,
+  onSubmit,
+  ...
+}) {
+  return ...
+}
+
+```
+
+## TypeScript
+
+### Handler event naming
+
+The naming of the handler method contains three parts.
+
+_handle_ _name of element_ _name of event_
+
+```typescript
+// bad
+const buttonclickHandler = (event) => ...
+
+// bad
+const handleClick = (event) => ...
+
+// good
+const handleButtonClick = (event) => ...
+
+// else good
+const handleFormSubmit = (event) => ...
+```
+
+### Variable Naming
+
+boolean variables staring with prefix `is`, `should`, `has`, `can`, `did`, `will`.
```

Comment:
please change to: `Boolean variables should start with one of these prefixes: ...`